### PR TITLE
style: naming examples

### PR DESCRIPTION
**Pull request by Daanvdplas**
There was a warning in compiling a contract when adding pop api dependency. This should resolve it.

I also updated the naming, which we can use to move forward with the examples.

This is just naming things, examples need a lot more work.

### CODE REVIEW

**Comment by evilrobot-01** [comment]
pop-api is included in the path, so having it in the contract names seems redundant. Simpler to just remove the `pop_api_` prefixes imo.

File: pop-api/examples/balance-transfer/lib.rs [hunk 1]
```diff
@@ -15,17 +15,17 @@ impl From<balances::Error> for ContractError {
 }
 
 #[ink::contract(env = pop_api::Environment)]
-mod pop_api_extension_demo {
+mod pop_api_balances {
```
Comment (Daanvdplas) [inline pop-api/examples/balance-transfer/lib.rs:18]: This couldn't be balances due to `pop_api::balances`.

Suggest we have a closer look at those names when adding examples properly working on pop api

File: pop-api/examples/balance-transfer/Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "pop_api_extension_demo"
+name = "balances"
```
Comment (brunopgalvao) [inline pop-api/examples/balance-transfer/Cargo.toml:2]: I would call it `balance_transfer`

File: pop-api/examples/balance-transfer/lib.rs (hunk 1 above)
Comment (brunopgalvao) [inline pop-api/examples/balance-transfer/lib.rs:18]: `mod balance_transfer`

File: pop-api/examples/balance-transfer/lib.rs [hunk 3]
```diff
@@ -15,17 +15,17 @@ impl From<balances::Error> for ContractError {
...
 	use super::ContractError;
 
 	#[ink(storage)]
 	#[derive(Default)]
-	pub struct PopApiExtensionDemo;
+	pub struct PopApiBalances;
```
Comment (brunopgalvao) [inline pop-api/examples/balance-transfer/lib.rs:23]: `BalanceTransfer`

File: pop-api/examples/nfts/lib.rs [hunk 4]
```diff
@@ -18,17 +18,17 @@ impl From<nfts::Error> for ContractError {
 }
 
 #[ink::contract(env = pop_api::Environment)]
-mod pop_api_extension_demo {
+mod pop_api_nfts {
```
Comment (brunopgalvao) [inline pop-api/examples/nfts/lib.rs:21]: `nfts`

File: pop-api/examples/nfts/lib.rs [hunk 5]
```diff
@@ -18,17 +18,17 @@ impl From<nfts::Error> for ContractError {
...
 	use super::ContractError;
 
 	#[ink(storage)]
 	#[derive(Default)]
-	pub struct PopApiExtensionDemo;
+	pub struct PopApiNfts;
```
Comment (brunopgalvao) [inline pop-api/examples/nfts/lib.rs:26]: `Nfts`

**Review by brunopgalvao — COMMENTED** [review]
In general, I would remove the `pop_api_` or `PopApi` prefix and just name it according to the name of the top level folder.

Similar to https://github.com/paritytech/ink-examples

File: pop-api/examples/place-spot-order/lib.rs [hunk 6]
```diff
@@ -1,34 +1,16 @@
...
-impl From<nfts::Error> for ContractError {
-	fn from(value: nfts::Error) -> Self {
-		ContractError::NftsError(value)
-	}
-}
-
 #[ink::contract(env = pop_api::Environment)]
-mod pop_api_spot_order_example {
-	use super::ContractError;
+mod spot_order {
```
Comment (Daanvdplas) [inline pop-api/examples/place-spot-order/lib.rs:4]: I removed the error handling. This should be implemented properly in another PR if we see value in this example (e.g. as a sub0 presentation reference).

File: pop-api/examples/balance-transfer/lib.rs [hunk 7]
```diff
@@ -1,49 +1,49 @@
...
 		#[ink(message)]
-		pub fn transfer_through_runtime(
+		pub fn transfer(
 			&mut self,
 			receiver: AccountId,
 			value: Balance,
 		) -> Result<(), ContractError> {
 			ink::env::debug_println!(
-				"PopApiExtensionDemo::transfer_through_runtime: \nreceiver: {:?}, \nvalue: {:?}",
+				"BalanceTransfer::transfer_through_runtime: \nreceiver: {:?}, \nvalue: {:?}",
```
Comment (brunopgalvao) [inline pop-api/examples/balance-transfer/lib.rs:39]: `"BalanceTransfer::transfer: \nreceiver: {:?}, \nvalue: {:?}",`

File: pop-api/examples/balance-transfer/Cargo.toml (hunk 2 above)
Comment (brunopgalvao) [inline pop-api/examples/balance-transfer/Cargo.toml:2]: I would call it `balance_transfer`. Same as top-level folder.

What I would do is actually change the top-level folder to be `balances` instead of `balance-transfer`.

File: pop-api/examples/balance-transfer/lib.rs [hunk 8]
```diff
@@ -1,49 +1,49 @@
...
 				receiver,
 				value
 			);
 
-			pop_api::balances::transfer_keep_alive(receiver, value)?;
+			transfer_keep_alive(receiver, value)?;
 
-			ink::env::debug_println!("PopApiExtensionDemo::transfer_through_runtime end");
+			ink::env::debug_println!("BalanceTransfer::transfer_through_runtime end");
```
Comment (brunopgalvao) [inline pop-api/examples/balance-transfer/lib.rs:46]: ditto

File: pop-api/examples/nfts/lib.rs [hunk 9]
```diff
@@ -40,34 +40,34 @@ mod pop_api_extension_demo {
 			receiver: AccountId,
 		) -> Result<(), ContractError> {
 			ink::env::debug_println!(
-				"Contract::mint_through_runtime: collection_id: {:?} item_id {:?} receiver: {:?}",
+				"Nfts::mint_through_runtime: collection_id: {:?} item_id {:?} receiver: {:?}",
```
Comment (brunopgalvao) [inline pop-api/examples/nfts/lib.rs:43]: Since you changed `transfer_through_runtime` in the balance example, you should change the function name here from `mint_through_runtime` to `mint` and the corresponding log messages.

**Review by al3mart — APPROVED** [review]
Looks good.
CI is failing but haven't spent much time figuring why

**Review by brunopgalvao — APPROVED** [review]

**Comment by Daanvdplas** [comment]
Closed by #86
